src/components/Divider.tsx

```diff
@@ -1,39 +1,34 @@
 import classNames from 'classnames'
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useDispatch } from 'react-redux'
 import Path from '../@types/Path'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
 import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import useMaxSiblingWidth from '../hooks/useMaxSiblingWidth'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import parentOf from '../util/parentOf'
 
 /** A custom horizontal rule. */
 const Divider = ({ path }: { path: Path }) => {
   const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
```

Comment:
I think you can replace `dividerSetWidth`, `width/setWidth`, and the `useEffect` with a single `useMemo` that depends on `maxSiblingWidth`.